tests/tracer/test_propagation.py

```diff
@@ -368,6 +373,305 @@ def test_get_wsgi_header(tracer):
+            "dd=o:rum;t.dm:-4;t.usr.id:baz64",
+            # sampling_priority_ts, meta, origin
+            (
+                None,
+                {
+                    "tracestate": "dd=o:rum;t.dm:-4;t.usr.id:baz64",
+                    "_dd.p.dm": "-4",
+                    "_dd.p.usr.id": "baz64",
+                },
+                "rum",
+            ),
+            None,
+        ),
+        (
+            "tracestate_with_negative_sampling_priority",
+            "dd=s:-1;o:rum;t.dm:-4;t.usr.id:baz64",
+            # sampling_priority_ts, meta, origin
+            (
+                -1,
+                {
+                    "tracestate": "dd=s:-1;o:rum;t.dm:-4;t.usr.id:baz64",
+                    "_dd.p.dm": "-4",
+                    "_dd.p.usr.id": "baz64",
+                },
+                "rum",
+            ),
+            None,
+        ),
+        (
+            "tracestate_with_no_origin",
+            "dd=s:2;t.dm:-4;t.usr.id:baz64",
+            # sampling_priority_ts, meta, origin
+            (
+                2,
+                {
+                    "tracestate": "dd=s:2;t.dm:-4;t.usr.id:baz64",
+                    "_dd.p.dm": "-4",
+                    "_dd.p.usr.id": "baz64",
+                },
+                None,
+            ),
+            None,
+        ),
+        (
+            "tracestate_with_unknown_t._values",
+            "dd=s:2;o:rum;t.dm:-4;t.usr.id:baz64;t.unk:unk,congo=t61rcWkgMzE,mako=s:2;o:rum;",
+            # sampling_priority_ts, meta, origin
+            (
+                2,
+                {
+                    "tracestate": "dd=s:2;o:rum;t.dm:-4;t.usr.id:baz64;t.unk:unk,congo=t61rcWkgMzE,mako=s:2;o:rum;",
+                    "_dd.p.dm": "-4",
+                    "_dd.p.usr.id": "baz64",
+                    "_dd.p.unk": "unk",
+                },
+                "rum",
+            ),
+            None,
+        ),
+        (
+            "tracestate_with_no_dd_list_member",
+            "congo=t61rcWkgMzE,mako=s:2;o:rum;",
+            # sampling_priority_ts, meta, origin
+            (
+                None,
+                {
+                    "tracestate": "congo=t61rcWkgMzE,mako=s:2;o:rum;",
+                },
+                "",
+            ),
+            ["no dd list member in tracestate from incoming request: 'congo=t61rcWkgMzE,mako=s:2;o:rum;'"],
+        ),
+        (
+            "tracestate_no_origin",
+            "dd=s:2;t.dm:-4;t.usr.id:baz64,congo=t61rcWkgMzE,mako=s:2;o:rum;",
+            # sampling_priority_ts, meta, origin
+            (
+                2,
+                {
+                    "tracestate": "dd=s:2;t.dm:-4;t.usr.id:baz64,congo=t61rcWkgMzE,mako=s:2;o:rum;",
+                    "_dd.p.dm": "-4",
+                    "_dd.p.usr.id": "baz64",
+                },
+                None,
+            ),
+            None,
+        ),
+    ],
+)
+def test_extract_tracestate(caplog, name, ts_string, expected_tuple, expected_logging):
```

Comment:
Can we add a test case with non ascii characters (ex: put emojis in the header) and a test where `dd=......,` has an invalid format (ex: `dd=foo|bar:hi|low$candy;:emoji`).